### PR TITLE
BGL: Add adjacency_iterator for several classes

### LINKED ISSUE #6531: Missing adjacency_iterator in some boost::graph_traits specializations
## Issue Details

According to https://doc.cgal.org/latest/BGL/group__PkgBGLTraits.html the specializations for  `boost::graph_traits` contain a member type `adjacency_iterator` (except for `CGAL::Arrangement_2`), but this is not the case  at least (I have not checked all) for `CGAL::Polyhedron_3`, `CGAL::Surface_Mesh` and `CGAL::Linear_cell_complex_for_combinatorial_map`.

The following code can be used to test.
```cpp
#include <CGAL/Simple_cartesian.h>
#include <CGAL/Polyhedron_3.h>

template <class T> bool test = true;

using Kernel = CGAL::Simple_cartesian<double>;
using Polyhedron = CGAL::Polyhedron_3<Kernel>;

int main() {
  test<typename boost::graph_traits<Polyhedron>::adjacency_iterator>; // does not compile
  test<typename boost::graph_traits<Polyhedron>::vertex_descriptor>; // compiles
}
```

## Environment

* Operating system Linux 64bits (It is missing in source code, so all platforms)
* Compiler: gcc (same remark, so all compilers)
* CGAL version: 5.4
* Boost version: 1.71.0 (most likely for all versions)

## Solution?

The triangulation classes do have a specialization of `boost::graph_traits` with a member type `adjacency_iterator`, so to solve this issue, one could look at that.

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Add `adjacency_iterator` to several classes with `typedef`s as defined in the [documentation](https://doc.cgal.org/latest/BGL/group__PkgBGLTraits.html).

Do we want to provide it for `Dual` etc ?

## Release Management

* Affected package(s): Surface_mesh, Polyhedron, LCC
* Issue(s) solved (if any): fix #6531

### CODE REVIEW

File: BGL/include/CGAL/boost/graph/graph_traits_OpenMesh.h [hunk 1]
```diff
@@ -264,6 +268,19 @@ out_edges(typename boost::graph_traits<OPEN_MESH_CLASS >::vertex_descriptor v,
 }
 
 
+template <typename K>
+CGAL::Iterator_range<typename boost::graph_traits<OPEN_MESH_CLASS >::adjacency_iterator>
+adjacnt_vertices(typename boost::graph_traits<OPEN_MESH_CLASS >::vertex_descriptor v,
```
Comment (sloriot) [inline BGL/include/CGAL/boost/graph/graph_traits_OpenMesh.h:273]:
```suggestion
adjacent_vertices(typename boost::graph_traits<OPEN_MESH_CLASS >::vertex_descriptor v,
```

Comment (afabri) [reply BGL/include/CGAL/boost/graph/graph_traits_OpenMesh.h]: fixed.

**Comment by sloriot** [comment]
Test failing [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-5.5-Ic-80/BGL/TestReport_lrineau_Fedora-rawhide.gz)

**Comment by sloriot** [comment]
Successfully tested in CGAL-5.5-Ic-83

**Pull request merged**